report/unsafe/sections/albin_arrowhead_communication.tex

```diff
@@ -0,0 +1,92 @@
+To enable intra cloud communication both the gateway and gatekeeper core systems will have to be running as well. 
+
+\subsection{Communication}
+This project uses the arrowhead framework for Communication between the robot and the local arrowhead cloud as well as the 
+remote cloud from the factory. This section of the report aims to explain how the communication between these different clouds works as well
+as to define the purpose of using the arrowhead framework. This project decided to use arrowhead framework for communication between 
+the local cloud and the robot, all internal communication on the robot will be handled using ROS. This proides faster and more robust 
+control of the robot, not relying on internet connection and so forth to control the robot. It should be noted that the arrowhead framework
+is fully competent to control every aspect of the robot as well, but can run in to problems with delay in time critical sections.
+
+\subsubsection{Intra cloud communication}
+The prerequisites for using intra cloud communication is:
+\begin{itemize}
+    \item Running core services i.e. service registry, authorization and orchestrator.
+    \item A registered consumer and provider.
+    \item Defined intra cloud rules.
+\end{itemize}
+
+\subsubsection{Inter cloud communication}
+The prerequisites for using inter cloud communication is:
+\begin{itemize}
+    \item Running core services i.e. service registry, authorization, orchestrator, gateway and gatekeeper.
+    \item A running ActiveMQ relay server.
+    \item Defined inter cloud rules.
+\end{itemize}
+
+\subsubsection{Purpose of using arrowhead framework}
+The communication between different IoT-devices has been an issue for quite some time, and the security issues surrounding IoT-devices
+are well known. Arrowhead framework attempts to provide a soulution for fixing these issues. Using HTPPS with certificates the arrowhead
+framework provides a solution for tackling the basic secruity flaws of oth er unprotected IoT-devices, this means that you would have to have
+the right certificate and the keyword for that certificate in order to access the service registry. The same applies to both the orchestrator
+and authorization core system. 
+
+\subsubsection{Services}
+Two services will be offered by the robot, which acts as a provider i.e. it provides a service. In this case the services offered
+is place and pick-up. Both these services will have to be registered in the service registry where they will have differnt URI:s 
+to and servicedefinitions to tell them apart. Otherwise they will work very much the same. They will, when prompted to do so, provide a
+service to either pick-up the piece from the end of the factorys conveyer belt or place it at the beginning of the conveyer belt. 
+In both cases the services will wait for a JSON containing if the service should be carried out or not. When recieving a JSON containing that the service will be carried out. Both the services
```

Comment:
"When recieving a JSON containing that the service will be carried out"
containing that the sound little bit forced.
mig want to write : containing the command the service...